src/client/QXmppAtmManager.cpp

```diff
@@ -0,0 +1,303 @@
+/*
+ * Copyright (C) 2008-2021 The QXmpp developers
+ *
+ * Author:
+ *  Melvin Keskin <melvo@olomono.de>
+ *
+ * Source:
+ *  https://github.com/qxmpp-project/qxmpp
+ *
+ * This file is a part of QXmpp library.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ */
+
+#include "QXmppAtmManager.h"
+
+#include <QDomElement>
+#include <QList>
+
+#include "QXmppClient.h"
+#include "QXmppConstants_p.h"
+#include "QXmppTrustMessageElement.h"
+#include "QXmppUtils.h"
+
+///
+/// \class QXmppAtmManager
+///
+/// \brief The QXmppAtmManager class represents a manager
+/// for \xep{0450} Automatic Trust Management (ATM).
+///
+/// \since QXmpp 1.5
+///
+
+///
+/// Constructs an ATM manager.
+///
+QXmppAtmManager::QXmppAtmManager()
+{
+}
+
+///
+/// Authenticates keys manually (e.g., by the Trust Message URI of a QR code).
+///
+/// \param encryption namespace of the encryption protocol that uses the keys
+/// \param keyOwnerJid JID of the key owner
+/// \param keyIds IDs of the keys being authenticated
+///
+void QXmppAtmManager::authenticateManually(const QString &encryption, const QString &keyOwnerJid, const QList<QString> &keyIds)
+{
+    authenticate(encryption, keyOwnerJid, keyIds, true);
+}
+
+///
+/// Distrusts keys manually (e.g., by the Trust Message URI of a QR code).
+///
+/// \param encryption namespace of the encryption protocol that uses the keys
+/// \param keyOwnerJid JID of the key owner
+/// \param keyIds IDs of the keys being distrusted
+///
+void QXmppAtmManager::distrustManually(const QString &encryption, const QString &keyOwnerJid, const QList<QString> &keyIds)
+{
+    distrust(encryption, keyOwnerJid, keyIds, true);
+}
+
+void QXmppAtmManager::setClient(QXmppClient *client)
+{
+    QXmppClientExtension::setClient(client);
+    connect(client, &QXmppClient::messageReceived, this, &QXmppAtmManager::handleMessageReceived);
+}
+
+void QXmppAtmManager::handleMessageReceived(const QXmppMessage &message)
+{
+    if (const auto trustMessageElement = message.trustMessageElement()) {
+        const bool isTrustMessageIntendedForAtm = trustMessageElement->usage() == ns_atm;
+
+        if (isTrustMessageIntendedForAtm) {
```

Comment:
I guess other cases are still coming? Otherwise please just put the condition into the if statement.